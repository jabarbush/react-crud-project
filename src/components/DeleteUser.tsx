import React, { useState } from 'react';
import DeleteUserProps from '../interfaces/DeleteUserProps';
import { ReactComponent as FormClose } from '../assets/form-close.svg';

const DeleteUser: React.FC<DeleteUserProps> = ({ userList, selectedUser, onClose, onDeleteUser }) => {
  const [isPopupVisible, setIsPopupVisible] = useState(true);

  const handleClosePopup = () => {
    setIsPopupVisible(false);
  };

  if (!isPopupVisible) {
    return null;
  }

  const handleDeleteUser = () => {
    const indexToRemove = userList.findIndex(obj => obj.id === selectedUser?.id);
    if (indexToRemove !== -1) {
        userList.splice(indexToRemove, 1);
      }
    setIsPopupVisible(false);
    onDeleteUser();
  };
//className="add-user-popup"
  return (
    <div className='delete-user-popup'>
        <h2 className="delete-user-title">Confirm Delete</h2>
        <FormClose onClick={handleClosePopup} className='icons icon' />
        <div className='delete-message'>Are you sure you want to delete?</div>
        <button className="cancel" onClick={handleClosePopup}>
          Cancel
        </button>
        <button className="add-user-submit" onClick={handleDeleteUser} >
          OK
        </button>
    </div>
  );
};

export default DeleteUser;
