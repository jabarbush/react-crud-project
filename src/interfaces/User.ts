interface User {
    id: number;
    name: string;
    dob: string;
    phone: string;
    email: string;
    signature?: {
        pin: string;
        fontStyle: string;
    } | null;
  }
  
  export default User;
  