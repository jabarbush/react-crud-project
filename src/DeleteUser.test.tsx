import { render, screen, fireEvent } from '@testing-library/react';
import User from './interfaces/User';
import DeleteUser from './components/DeleteUser';

beforeEach(() => jest.resetAllMocks())
describe('DeleteUser component', () => {
    const selectedUser = {
        id: 1,
        name: 'jake',
        dob: '03/03/2003',
        phone: '4564564564',
        email: 'jake4@gmail.com',
        signature: {
          pin: '123123',
          fontStyle: 'Mr Dafoe',
        },
      };
    const userList: User[] = [{
        id: 1,
        name: 'jake',
        dob: '03/03/2003',
        phone: '4564564564',
        email: 'jake4@gmail.com',
        signature: {
            pin: '123123',
            fontStyle: 'Mr Dafoe',
        },
    },
    {
        "id": 2,
        "name": "jake425525333",
        "dob": "03/03/2003",
        "phone": "4564564564",
        "email": "jake4@gmail.com",
        "signature": null
      },
      {
        "id": 3,
        "name": "ashley1234",
        "dob": "05/05/1995",
        "phone": "0980980988",
        "email": "ashley@jake.com",
        "signature": {
          "pin": "123123",
          "fontStyle": "Dancing Script"
        }
      },
      {
        "id": 4,
        "name": "123",
        "dob": "01/01/2001",
        "phone": "1231231231",
        "email": "jake@gmail.com",
        "signature": {
          "pin": "123123",
          "fontStyle": "Great Vibes"
        }
      }];

    const onCloseUserInfo = jest.fn();
    const setUserList = jest.fn();
    const handleOpenDeleteUser = jest.fn();

    test('Delete component renders', () => {
        render(
            <DeleteUser
                userList={userList}
                selectedUser={selectedUser}
                setUserList={setUserList}
                onClose={onCloseUserInfo}
                onDeleteUser={handleOpenDeleteUser}
            />
        );
        const deleteDialogElement = screen.getByText('Confirm Delete');
        expect(deleteDialogElement).toBeInTheDocument();
    });

    //i know, not ideal
    test('Clicking OK removes selected user from the user list', () => {
        render(
          <DeleteUser
            userList={userList}
            selectedUser={selectedUser}
            setUserList={setUserList}
            onClose={onCloseUserInfo}
            onDeleteUser={handleOpenDeleteUser}
          />
        );
      
        const okButton = screen.getByText('OK');
        fireEvent.click(okButton);
        const updatedUserList = userList.filter(user => user.id !== selectedUser.id);
        render(
          <DeleteUser
            userList={updatedUserList}
            selectedUser={null}
            setUserList={setUserList}
            onClose={onCloseUserInfo}
            onDeleteUser={handleOpenDeleteUser}
          />
        );
      
        const deletedUser = screen.queryByText(selectedUser.name);
        expect(deletedUser).not.toBeInTheDocument();
      });
      

      test('DeleteUser component matches snapshot', () => {
        
        const onClose = jest.fn();
        const onDeleteUser = jest.fn();
        const setUserList = jest.fn();
      
        const { asFragment } = render(
          <DeleteUser
            userList={userList}
            selectedUser={selectedUser}
            onClose={onClose}
            onDeleteUser={onDeleteUser}
            setUserList={setUserList}
          />
        );
      
        expect(asFragment()).toMatchSnapshot();
      });
});