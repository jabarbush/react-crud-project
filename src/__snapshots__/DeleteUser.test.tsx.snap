// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteUser component DeleteUser component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="delete-user-overlay"
  >
    <div
      class="delete-user-popup"
    >
      <h2
        class="delete-user-title"
      >
        Confirm Delete
      </h2>
      <svg
        class="icons icon-big-x"
        data-testid="delete-x-button"
      >
        form-close.svg
      </svg>
      <div
        class="delete-message"
      >
        Are you sure you want to delete?
      </div>
      <button
        class="cancel"
        name="cancel"
      >
        Cancel
      </button>
      <button
        class="add-user-submit"
      >
        OK
      </button>
    </div>
  </div>
</DocumentFragment>
`;
